llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -36,8 +36,8 @@ VPTypeAnalysis::VPTypeAnalysis(const VPlan &Plan) : Ctx(Plan.getContext()) {
   // If there's no canonical IV, retrieve the type from the trip count
   // expression.
   auto *TC = Plan.getTripCount();
-  if (TC->isLiveIn()) {
-    CanonicalIVTy = TC->getLiveInIRValue()->getType();
+  if (auto *TCLI = dyn_cast<VPIRValue>(TC)) {
```

Comment:
Here and elsewhere `LI` stands for LiveIn. Wonder if there's a better short name for VPIRValue.